src/migrations/20220203152515-bootstrap_accounts.ts

```diff
@@ -0,0 +1,90 @@
+/* eslint @typescript-eslint/no-var-requires: "off" */
+// @ts-expect-error ts complain about "Cannot redeclare block-scoped variable 'crypto'"
+const crypto = require("crypto")
+
+const adminUsers = [
+  {
+    phone: "+16505554327",
+    role: "dealer",
+    needUsdWallet: true,
+  },
+  { phone: "+16505554325", role: "funder" },
+  { phone: "+16505554334", role: "bankowner" },
+]
+
+const getRandomInvalidPhone = () => {
+  return `+abc${Math.floor(Math.random() * 999_999_999_999_999)}`
+}
+
+const defaultWallet = { __v: 0, type: "checking", onchain: [] }
+
+module.exports = {
+  async up(db /*, client */) {
+    for (const userMeta of adminUsers) {
+      const role = userMeta.role
+      let user = await db.collection("users").findOne({ role })
+
+      let userId
+      if (!user) {
+        const phone = getRandomInvalidPhone()
+        const created_at = new Date()
+        const result = await db.collection("users").insertOne({ role, phone, created_at })
+        userId = result.insertedId
+
+        user = await db.collection("users").findOne({ role })
+      } else {
+        console.log({ role }, "user exist")
+        userId = user._id
+      }
+
+      const _accountId = userId
+      // const accountId = Types.ObjectId(userId)
+      let wallet = await db.collection("wallets").findOne({ _accountId, currency: "BTC" })
+
+      if (!wallet) {
+        const currency = "BTC"
+        // @ts-expect-error: ts complain about Crypto (capital C) doesn't have randomUUID
+        const id = crypto.randomUUID()
+
+        const result = await db
+          .collection("wallets")
+          .insertOne({ _accountId, currency, id, ...defaultWallet })
+
+        wallet = await db.collection("wallets").findOne({ _accountId, currency: "BTC" })
+
+        console.log({ result, currency }, "wallet has been created")
+      } else {
+        console.log({ role, walletExist: wallet }, "wallet exist")
+      }
+
+      if (user.defaultWalletId === undefined) {
+        const result = await db
+          .collection("users")
+          .findOneAndUpdate({ role }, { $set: { defaultWalletId: wallet.id } })
+
+        console.log({ result, role }, "update defaultWalletId")
+      } else {
+        console.log({ role }, "no need to update defaultWalletId")
+      }
+
+      if (userMeta.needUsdWallet) {
+        const currency = "USD"
+        wallet = await db.collection("wallets").findOne({ _accountId, currency })
+
+        if (!wallet) {
+          // @ts-expect-error: ts complain about Crypto (capital C) doesn't have randomUUID
+          const id = crypto.randomUUID()
+          const result = await db
+            .collection("wallets")
+            .insertOne({ _accountId, currency, id, ...defaultWallet })
+
+          wallet = await db.collection("wallets").findOne({ _accountId, currency: "BTC" })
```

Comment:
What's the purpose of this line?

Reply:
check if a wallet already exists. we don't do anything in that case.

Reply:
this script will be used both for new cluster, but also existing cluster that have those users (partially) setup